llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -4977,8 +4977,6 @@ const Value *getAddress(const DbgVariableIntrinsic *DVI) {
 }
 
 const Value *getAddress(const DbgVariableRecord *DVR) {
-  assert(DVR->getType() == DbgVariableRecord::LocationType::Declare ||
-         DVR->getType() == DbgVariableRecord::LocationType::Assign);
```

Comment:
Removing this assert is questionable imo - although `getAddress()` on a dbg_value will currently give you the location operand, it's not guaranteed to work and will likely be broken by future API changes.

Reply:
Hm, well I said that removing the assert was questionable, but the problem is really calling `DVR->getAddress()` for a dbg_value; at some point, we intend to refactor the classes s.t. `getAddress()` is not available for dbg_values. I think the future-proof code would look more like:
```
  if (DVR->isDbgValue())
    return DVR->getValue();
  return DVR->getAddress();
```
But since there's no error with just using `DVR->getAddress()` right now, I think this can be left for us to fix when we refactor this.